src/library_wasi.js

```diff
@@ -11,19 +11,31 @@ var WasiLibrary = {
     return _exit(code);
   },
 
+  // This object can be modified by the user during startup, which affects
+  // the initial values of the environment accessible by getenv.
+  $ENV: {},
+
+  emscripten_get_environ__deps: ['$ENV'],
   emscripten_get_environ: function() {
     if (!_emscripten_get_environ.strings) {
-      var ENV = {};
-      ENV['USER'] = ENV['LOGNAME'] = 'web_user';
-      ENV['PATH'] = '/';
-      ENV['PWD'] = '/';
-      ENV['HOME'] = '/home/web_user';
-      // Browser language detection #8751
-      ENV['LANG'] = ((typeof navigator === 'object' &&navigator.languages && navigator.languages[0]) || 'C').replace('-', '_') + '.UTF-8';
-      ENV['_'] = thisProgram;
+      // Default values.
+      var env = {
+        'USER': 'web_user',
+        'LOGNAME': 'web_user',
```

Comment:
Why add LOGNAME here?

Reply:
It was there before too, just on a single line,
```
ENV['USER'] = ENV['LOGNAME'] = 'web_user';
```